pkg/packageinstall/packageinstall.go

```diff
@@ -301,7 +301,18 @@ func (pi *PackageInstallCR) referencedPkgVersion() (datapkgingv1alpha1.Package,
 
 	// we only need to populate the versionInfo we know that the packages have constraints that will require this info.
 	if requiresClusterVersion {
-		v, err := pi.compInfo.KubernetesVersion(pi.model.Spec.ServiceAccountName, pi.model.Spec.Cluster, &pi.model.ObjectMeta)
+		kubernetesVersion := func() (semver.Version, error) {
+			memoizedKubernetesVersion := semver.Version{}
```

Comment:
here the naming suggests you're memoizing but the variable doesn't have scope outside the function, right?  am I missing how this would be persisted across function calls?   (vs. a couple files up where you declared the variables in the outer scope and then assigned to them in the func)